diff --git a/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp b/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp
index 190e9e10e..2a6e2893d 100644
--- a/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp
+++ b/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp
@@ -189,8 +189,14 @@ MIGRAPHX_DEVICE_MATH_BINARY_FOR(float, max, ::fmaxf)
 MIGRAPHX_DEVICE_MATH_BINARY_FOR(float, min, ::fminf)
 MIGRAPHX_DEVICE_MATH_BINARY_FOR(double, max, ::max)
 MIGRAPHX_DEVICE_MATH_BINARY_FOR(double, min, ::min)
-MIGRAPHX_DEVICE_MATH_BINARY_FOR(migraphx::half, max, ::__hmax)
-MIGRAPHX_DEVICE_MATH_BINARY_FOR(migraphx::half, min, ::__hmin)
+inline auto __device__ max(migraphx::half x, migraphx::half y) -> migraphx::half
+{
+    return static_cast<migraphx::half>(::__hmax(static_cast<__half>(x), static_cast<__half>(y)));
+}
+inline auto __device__ min(migraphx::half x, migraphx::half y) -> migraphx::half
+{
+    return static_cast<migraphx::half>(::__hmin(static_cast<__half>(x), static_cast<__half>(y)));
+}
 
 template <class T, MIGRAPHX_REQUIRES(not is_any_vec<T>() and is_integral<T>{})>
 constexpr auto abs(const T& a)
